### ISSUE TITLE
Browser not launching

### ISSUE THREAD

**Issue by marcusdiy** [CheshireCaat/puppeteer-with-fingerprints#62]
Im getting many errors from this request: https://bablosoft.com/scripts/FingerprintPluginV5/properties
The page is not loading.

I can get the info using another vpn: {"success":true,"free":true,"hash":"1713cb035fbd3cfebeff363309e3472ffe469f61","engversion":"26.2.0"}

So no I have two questions?
1) Why this request, pretty usless? If not as a killswitch
2) Is there some kind of rate limiting

**Comment by bablosoft** [comment]
This could be because your ip has been banned.

Old version starts BAS instance on each profile start.

The ban is temporary, need to wait while it would be cancelled and try again.

Please also ensure that you are using latest version and don't start node process per profile launch.

**Comment by marcusdiy** [comment]
I have to start a node process per launch because multithreading has lock errors

**Comment by bablosoft** [comment]
Please check this example on how multithreading can be implemented.

https://github.com/CheshireCaat/puppeteer-with-fingerprints/blob/master/examples/multithreading.js

I've tested it many times, it is fully functional.

If you still have encountered any error, please create new issue with minimal source, how it could be reproduced.

**Comment by marcusdiy** [comment]
Will test you example again.
What are the requirements to ban an ip? How many requests?
What is the difference between multithreading and launching multiple node.js instances? 
Beside a couple of requests more to start the browsers in parallel... is there a parallel license to buy?

**Comment by bablosoft** [comment]
> What are the requirements to ban an ip? How many requests?

I can't say that.

> What is the difference between multithreading and launching multiple node.js instances?

When using multithreading only single BAS instance is started, otherwise new BAS instance is started for each node instance.

> Beside a couple of requests more to start the browsers in parallel... is there a parallel license to buy?

No

**Comment by marcusdiy** [comment]
Im ok with multiple BAS instances running

**Comment by bablosoft** [comment]
Each BAS launch is very expensive on server side. Please use multithreaded architecture instead of multiprocess.

**Comment by marcusdiy** [comment]
I sure will pay for better fingerprints, sure may pay for some ip analysis for proxies but paying for an unknown process that may fail on me at any time? Doesnt seem a good idea... what processes are required on server side? Can i host it myself?

**Comment by bablosoft** [comment]
Right now the only option is to use single node process. This will ensure that you are not getting banned despite how many profiles you will run. This option is free, you don't need to pay for it. Other options are not available.

**Comment by marcusdiy** [comment]
There are always other options... a delay will do for now.

**Comment by bablosoft** [comment]
> a delay will do for now.

Ok, but I must warn, that would lead to hurting server stability and as a consequence may lead to blocking your ips/account.

**Comment by marcusdiy** [comment]
I will start them every 3 minutes, should be a problem, no? 
Btw the other bots are a copy of the first folder so there is no download occuring. 
Will monitor what kind of traffic is send to BAS just out of curiosity.

**Comment by bablosoft** [comment]
> I will start them every 3 minutes, should be a problem, no?

If you start 1 profile per 3 minutes - it is ok.

If you start a lot of profiles - it would be a problem.
